### DISCUSSION TITLE
Prisma connecting to Postgresql on Google Cloud Functions with Node.js 18

### DISCUSSION THREAD

**Discussion by pip8786** [prisma/prisma#22520] [Q&A] [answered]
### Question

I've been running our Google Cloud Functions on Node.js 16 for a long time, but that is about to stop working at the end of January. I was hoping to just update to Node.js 18 which is what my machine runs and move on from there. When I updated, my cloud functions started reporting an error when trying to use prisma:

```
Error opening a TLS connection: error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:../crypto/evp/evp_fetch.c:349:Global default library context, Algorithm (RC2-40-CBC : 0), Properties () 
at si.handleRequestError (/workspace/node_modules/@prisma/client/runtime/library.js:125:7117) 
at si.handleAndLogRequestError (/workspace/node_modules/@prisma/client/runtime/library.js:125:6151) 
at si.request (/workspace/node_modules/@prisma/client/runtime/library.js:125:5859) 
at async l (/workspace/node_modules/@prisma/client/runtime/library.js:130:10025) 
```

I've been researching this all over the place and recommendations to downgrade to Node.js 16 or enable `--openssl-legacy-provider` aren't possible. Something I noticed is that Node.js 18 on GCP Functions runs on Ubuntu 22.04 while previous versions were on 18.04 (https://cloud.google.com/functions/docs/concepts/execution-environment#runtimes). I've seen a few places in the github issues mentioning problems on that OS but they seem to be closed. 

Any tips on how I can get my cloud functions connecting while running on Node.js 18?

### Information about Prisma Schema, Client Queries and Environment (optional)

DB URL is similar to: 
`postgres://${DB_USER}:${DB_PASS}@${DB_HOST}:${DB_PORT}/${DB_BASE}?sslidentity=${DB_ID}&sslpassword=${DB_ID_PASS}&sslcert=${DB_CERT}`

The sslidentity is a .p12 file and the sslcert is a .pem file. These both worked on the previous deployment. They also work on Node18 locally and on Vercel. Just can't figure it out on Google. 

- OS: Ubuntu 22.04
- Database: PostgreSQL on GCP using SSL certificates
- Node.js version: 18.18.2

- Prisma 5.7.1

**Comment by Jolg42** [comment]
Hi @pip8786

This doesn't look good, so when I saw your problem I tried to reproduce this, and "good news" I could reproduce this.

So here is what I tried,

1. Upgraded our `gcp-functions` test from Node.js v16 to v18
	We have a minimal deployment of Google Cloud Functions running on Node.js v16 called `gcp-functions`, and I upgraded the runtime to v18 in this PR:
	https://github.com/prisma/ecosystem-tests/pull/4433
	Note that this connects to an AWS RDS database.
	I can see in the logs that the Prisma engine switched, liked expected, from `libquery_engine-debian-openssl-1.0.x.so.node` to `libquery_engine-debian-openssl-3.0.x.so.node` and that the Prisma Client queries were successful.
	This was successful and working as expected

2. Upgraded our `debian-bullseye-amd64-openssl-1.1.x-tls` test and modified it as `debian-latest-amd64-openssl-3.0.x-tls`
	I did that in the same PR (https://github.com/prisma/ecosystem-tests/pull/4433)
	This is testing Prisma Client using GCP Postgres with the SSL certificate.
	And I can reproduce what you mentioned, so we need to turn this into a bug issue.
	Logs
	```
	PrismaClientInitializationError: 
	Invalid `client.user.findMany()` invocation in
	/usr/src/app/server.js:12:34
	
	   9 const client = new PrismaClient()
	  10 
	  11 app.get('/', async (req, res) => {
	→ 12   const data = await client.user.findMany(
	Error opening a TLS connection: error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:../crypto/evp/evp_fetch.c:373:Global default library context, Algorithm (RC2-40-CBC : 0), Properties ()
	    at si.handleRequestError (/usr/src/app/node_modules/.pnpm/@prisma+client@5.8.0-dev.50_prisma@5.8.0-dev.50/node_modules/@prisma/client/runtime/library.js:125:7117)
	    at si.handleAndLogRequestError (/usr/src/app/node_modules/.pnpm/@prisma+client@5.8.0-dev.50_prisma@5.8.0-dev.50/node_modules/@prisma/client/runtime/library.js:125:6151)
	    at si.request (/usr/src/app/node_modules/.pnpm/@prisma+client@5.8.0-dev.50_prisma@5.8.0-dev.50/node_modules/@prisma/client/runtime/library.js:125:5859)
	    at async l (/usr/src/app/node_modules/.pnpm/@prisma+client@5.8.0-dev.50_prisma@5.8.0-dev.50/node_modules/@prisma/client/runtime/library.js:130:9805)
	    at async /usr/src/app/server.js:12:16 {
	  clientVersion: '5.8.0-dev.50',
	  errorCode: undefined
	}
	```

**Reply by Jolg42** [reply, marked as answer]
Let's continue in your issue at https://github.com/prisma/prisma/issues/22568
